### DISCUSSION TITLE
Assert directive referencing an element of a tuple struct

### DISCUSSION THREAD

**Discussion by faro-dr** [jam1garner/binrw#231] [Q&A] [answered]
The following was working in version 0.11.2:

``` Rust
use binrw::binrw;

#[binrw]
#[bw(assert(self.0 > 0))]
pub struct u24(pub u32);
```

But fails in version 0.12.0 with:

````
error[E0425]: cannot find value `__binrw_this` in this scope
 --> src/main.rs:4:13
  |
4 | #[bw(assert(self.0 > 0))]
  |             ^^^^ not found in this scope
````

How can I refer to the elements of a tuple struct in an assert directive (if at all)? Thank you.

**Comment by csnover** [comment, marked as answer]
You can access tuple fields by the variable `self_N`, where `N` is the number of the field. However, it should be possible to also use `self` at the top level, so I have opened #232 to track this.

**Reply by faro-dr** [reply]
Thanks, that's a reasonable workaround.
